dbms/src/Functions/FunctionsGrouping.h

```diff
@@ -0,0 +1,232 @@
+// Copyright 2023 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Common/typeid_cast.h>
+#include <Core/ColumnNumbers.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Functions/FunctionHelpers.h>
+#include <Functions/IFunction.h>
+#include <Columns/ColumnNullable.h>
+#include <common/types.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+extern const int TOO_LESS_ARGUMENTS_FOR_FUNCTION;
+extern const int TOO_MANY_ARGUMENTS_FOR_FUNCTION;
+} // namespace ErrorCodes
+
+struct NameGrouping
+{
+    static constexpr auto name = "grouping";
+};
+
+template <typename Name>
+class FunctionGrouping : public IFunction
+{
+public:
+    using ResultType = UInt8;
+    using ArgType = UInt64; // arg type should always be UInt64
+    static constexpr auto name = Name::name;
+
+    static FunctionPtr create(const Context &) { return std::make_shared<FunctionGrouping>(); }
+    String getName() const override { return name; }
+    void setCollator(const TiDB::TiDBCollatorPtr & collator_) override { collator = collator_; }
+    bool useDefaultImplementationForNulls() const override { return false; }
```

Comment:
Can you breifly describe how the function deals with null values?

Reply:
> Can you breifly describe how the function deals with null values?

`useDefaultImplementationForNulls` is used for getting the nested column in a nullable column when it's true. However, this function is impossible to encounter a nullable column so it's needless to check nullable before entering this function.